awx/main/management/commands/host_metric.py

```diff
@@ -33,17 +174,39 @@ def handle(self, *args, **options):
         if until is not None:
             filter_kwargs['last_automation__lte'] = until
 
-        result = HostMetric.objects.filter(**filter_kwargs)
+        filter_kwargs_host_metrics_summary = {}
+        if since is not None:
+            filter_kwargs_host_metrics_summary['date__gte'] = since
+        if until is not None:
+            filter_kwargs_host_metrics_summary['date__lte'] = until
+
+        host_metric_row_count = HostMetric.objects.filter(**filter_kwargs).count()
+        host_metric_summary_monthly_row_count = HostMetricSummaryMonthly.objects.filter(**filter_kwargs_host_metrics_summary).count()
+
+        if (host_metric_row_count > PREFERRED_ROW_COUNT or host_metric_summary_monthly_row_count > PREFERRED_ROW_COUNT) and (
```

Comment:
The default value is already set to 500k.
We can set `--rows_per_file` to be an optional customization -- that's fine.

Note: This will require a change in the ADR

Reply:
I think ADR lists that as optional (stating 500k should be default)